tests/Fixer/Phpdoc/PhpdocListTypeFixerTest.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\Phpdoc;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\Phpdoc\PhpdocListTypeFixer
+ */
+final class PhpdocListTypeFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param array{style?: string} $configuration
+     *
+     * @dataProvider provideFixCases
+     */
+    public function testFix(string $expected, ?string $input = null, array $configuration = []): void
+    {
+        $this->fixer->configure($configuration);
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return iterable<array{string, 1?: null|string, 2?: array{style: string}}>
+     */
+    public static function provideFixCases(): iterable
+    {
+        yield ['<?php /** @tagNotSupportingTypes string[] */'];
+
+        yield ['<?php /** @var $variableWithoutType */'];
+
+        yield [
+            '<?php /** @var list<int> */',
+            '<?php /** @var int[] */',
+        ];
+
+        yield [
+            '<?php /** @param list<list<list<list<int>>>> $x */',
+            '<?php /** @param int[][][][] $x */',
+        ];
+
+        yield [
+            '<?php /** @return iterable<list<int>> */',
+            '<?php /** @return iterable<int[]> */',
+        ];
+
+        yield [
+            '<?php /** @var list<Foo\Bar> */',
+            '<?php /** @var Foo\Bar[] */',
+        ];
+
+        yield [
+            '<?php /** @var list<Foo_Bar> */',
+            '<?php /** @var Foo_Bar[] */',
+        ];
+
+        yield [
+            '<?php /** @var list<bool>|list<float>|list<int>|list<string> */',
+            '<?php /** @var array<bool>|float[]|array<int>|string[] */',
```

Comment:
`array<T>` should never be replaced with `list<T>` because they are not the same type.

Reply:
I get your point but I think this rule should only focus on syntax, i.e. `T[]` ⟷ `array<T>`. IMO forbidding `array<T>` is a matter of static analysis and out of scope here.

Reply:
But I can see the scenario when developer wants to apply fixer with `list` as an target type, then run SA, fix not matching entries and keep the correct ones. Ability to apply it with a tool would save a lot of time in cases where there's more actual `list`s than `array`s (comparing to applying `array<>` everywhere and converting to `list<>` manually).